packages/contracts/contracts/TellerV2.sol

```diff
@@ -353,9 +354,24 @@ contract TellerV2 is
         bid.loanDetails.loanDuration = _duration;
         bid.loanDetails.timestamp = uint32(block.timestamp);
 
-        bid.terms.paymentCycle = marketRegistry.getPaymentCycleDuration(
-            _marketplaceId
-        );
+        // Set payment cycle type based on market setting (custom or monthly)
+        bidPaymentCycleType[bidId] = marketRegistry
+            .getMarketplacePaymentCycleType(_marketplaceId);
+
+        if (
+            bidPaymentCycleType[bidId] ==
+            IMarketRegistry.PaymentCycleType.Seconds
+        ) {
+            bid.terms.paymentCycle = marketRegistry.getPaymentCycleValue(
+                _marketplaceId
+            );
+        } else if (
+            bidPaymentCycleType[bidId] ==
+            IMarketRegistry.PaymentCycleType.Monthly
+        ) {
+            bid.terms.paymentCycle = 30 days;
```

Comment:
If a monthly payment cycle should always land on the same day of the month (i.e. the 5th of each month)  why are we even bothering to instantiate a variable to be '30 days'.  What is 30 days?  Months are not 30 days long.

Reply:
the paymentCycle set in the bid terms is being used to calculate the payment cycle amount through the V2Calculations lib, which is why it's being set to 30 days (360 days/12 months for even monthly payments)